src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java

```diff
@@ -0,0 +1,75 @@
+package com.denizenscript.clientizen.events;
+
+import com.denizenscript.denizencore.events.ScriptEvent;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import net.minecraft.client.gui.screen.GameMenuScreen;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.advancement.AdvancementsScreen;
+import net.minecraft.client.gui.screen.ingame.CreativeInventoryScreen;
+import net.minecraft.client.gui.screen.ingame.InventoryScreen;
+import net.minecraft.client.gui.screen.option.OptionsScreen;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ScreenOpenCloseEvent extends ScriptEvent {
+
+	public static ScreenOpenCloseEvent instance;
+
+	public static Map<String, Class<?>> TYPE_MAP = new HashMap<>();
+
+	static {
+		TYPE_MAP.put("inventory", InventoryScreen.class);
+		TYPE_MAP.put("creative", CreativeInventoryScreen.class);
+		TYPE_MAP.put("pause", GameMenuScreen.class);
+		TYPE_MAP.put("options", OptionsScreen.class);
+		TYPE_MAP.put("advancements", AdvancementsScreen.class);
+	}
+
+	public String type;
+	public boolean opened;
+	public boolean switched;
+
+	public ScreenOpenCloseEvent() {
+		registerCouldMatcher("(inventory|creative|pause|options|advancements) screen opened|closed");
```

Comment:
Shouldn't the screen matcher be required (I.e. no `()`)? also I think you'd have `<'inventory'>` or `<'screen'>` as opposed to listing all types, and have all possible types in the meta (which we don't have yet, ofc)

Reply:
This is still unresolved

Reply:
I suppose due to the number of screens that will be matched, there's no point in having an optional.